Script updates from curation and database management work

Bankso opened the pull request.

Bankso committed:
Add shell=True to subprocess commands

Bankso committed:
Add shell=True to subprocess commands

Bankso committed:
Add initial version of build_workshop_project.py

Bankso committed:
Add initial version of build_workshop_project.py

Bankso committed:
Note authorship on other utils scripts

Bankso committed:
Note authorship on other utils scripts

Bankso committed:
Update/add function descriptions

Bankso committed:
Update/add function descriptions

Bankso committed:
Update to main wiki content

Reorder info
Add join team button, using team_id returned from create_team function
Add link to CCKP

Bankso committed:
Update to main wiki content

Reorder info
Add join team button, using team_id returned from create_team function
Add link to CCKP

Bankso committed:
Revert "Add shell=True to subprocess commands"

This reverts commit ed0f1b435f176f8349a4385a2969e5b99b78ed7c.

Bankso committed:
Revert "Add shell=True to subprocess commands"

This reverts commit ed0f1b435f176f8349a4385a2969e5b99b78ed7c.

Bankso committed:
Merge branch 'main' into workshop-builder

Bankso committed:
lint with black

Bankso committed:
lint with black

Bankso committed:
Update union_qc.py

- Add option for strict comparisons, which use additional columns to determine if updated entries are in the union table csv
- Formatting

Bankso committed:
Update union_qc.py

- Add option for strict comparisons, which use additional columns to determine if updated entries are in the union table csv
- Formatting

Bankso committed:
Create sync_manifest.py

Bankso committed:
Create sync_manifest.py

Bankso committed:
Create prep_backpopulation.py

Initial script reorders columns from database to union table ordering, when backpopulating from a database CSV
'add_missing_info' was used to add DOI URL information.
This script can be expanded as needed to incorporate functions needed for updating/formatting tables prior to backpopulation

Bankso committed:
Create prep_backpopulation.py

Initial script reorders columns from database to union table ordering, when backpopulating from a database CSV
'add_missing_info' was used to add DOI URL information.
This script can be expanded as needed to incorporate functions needed for updating/formatting tables prior to backpopulation

Bankso committed:
Update split_manifest_grants.py

- Add flag and configuration for working with manifests that will be used for backpopulation

Bankso committed:
Update split_manifest_grants.py

- Add flag and configuration for working with manifests that will be used for backpopulation

Bankso committed:
Update upload_validation.py

- Add flag and logic to skip CSV upload

Bankso committed:
Update upload_validation.py

- Add flag and logic to skip CSV upload

Bankso committed:
Create build_datasets.py

First commit of a simple script to create a Synapse Dataset that contains all files in a folder. Files can be in nested folders. This will be expanded to include logic for creating Datasets based on annotations, etc.

Bankso committed:
Create build_datasets.py

First commit of a simple script to create a Synapse Dataset that contains all files in a folder. Files can be in nested folders. This will be expanded to include logic for creating Datasets based on annotations, etc.

Bankso committed:
Update union_qc.py

Bankso committed:
Update union_qc.py

Bankso committed:
Update build_datasets.py

Bankso committed:
Update build_datasets.py

Bankso committed:
Create version_datasets.py

Bankso committed:
Create version_datasets.py

Bankso committed:
Allow datasets to be included in views

Bankso committed:
Allow datasets to be included in views

Bankso committed:
Add additional components for UNION tables

Bankso committed:
Add additional components for UNION tables

Bankso committed:
Create create_search_view.py

Adds a basic script for creating a materialized view table with metadata from different MC2 Center resources

Bankso committed:
Create create_search_view.py

Adds a basic script for creating a materialized view table with metadata from different MC2 Center resources

Bankso committed:
Create add_download_wiki.py

Adds a basic script and framework for adding content to a wiki. The wiki content is not generalized to multiple download situations, but can be adapted to fit other contexts.

Bankso committed:
Create add_download_wiki.py

Adds a basic script and framework for adding content to a wiki. The wiki content is not generalized to multiple download situations, but can be adapted to fit other contexts.

Bankso committed:
Create synapse_json_schema_bind.py

Add first version of script that binds a JSON schema to a Synapse entity. Note that schematic-derived JSON schemas are not inherently compatible with Synapse JSON services. Some elements of this script are hard coded for MC2-specific resources. This script does not add annotations to entities with bound schemas, which is being handled by a separate script in development

Bankso committed:
Create synapse_json_schema_bind.py

Add first version of script that binds a JSON schema to a Synapse entity. Note that schematic-derived JSON schemas are not inherently compatible with Synapse JSON services. Some elements of this script are hard coded for MC2-specific resources. This script does not add annotations to entities with bound schemas, which is being handled by a separate script in development

Bankso committed:
Create table_to_annotations.py

Adds a framework for accessing annotations in a Synapse table and applying them to a Synapse entity. This script assumes that the entity has previously had a schema bound to it.

Note: this script access annotations and attempts to apply them to the correct entity, but has not yet been used to apply annotations. There are some formatting issues in the JSON schemas generated by schematic that need to be accounted for and column names in Synapse tables must be adjusted to remove spaces, in order to serve as keys in the 'key:value' annotation pairs.

Bankso committed:
Create table_to_annotations.py

Adds a framework for accessing annotations in a Synapse table and applying them to a Synapse entity. This script assumes that the entity has previously had a schema bound to it.

Note: this script access annotations and attempts to apply them to the correct entity, but has not yet been used to apply annotations. There are some formatting issues in the JSON schemas generated by schematic that need to be accounted for and column names in Synapse tables must be adjusted to remove spaces, in order to serve as keys in the 'key:value' annotation pairs.

Bankso committed:
Update merge_tables.py

Add default Syn Ids for inputs

Bankso committed:
Update merge_tables.py

Add default Syn Ids for inputs

Bankso committed:
Create move_files.sh

Added a simple utility script for moving a list of files to a new folder

Bankso committed:
Create move_files.sh

Added a simple utility script for moving a list of files to a new folder

Bankso committed:
Create s3_handle_store.py

Added a simple script for 1) registering an AWS bucket with Synapse, 2) generating an S3 file handle from a file stored in the registered bucket and 3) storing the file handle in Synapse

Bankso committed:
Create s3_handle_store.py

Added a simple script for 1) registering an AWS bucket with Synapse, 2) generating an S3 file handle from a file stored in the registered bucket and 3) storing the file handle in Synapse

Bankso committed:
Merge branch 'workshop-builder' into add-utils-11-24

Bankso committed:
Update add_download_wiki.py

Add description to dataset wiki

Bankso committed:
Update add_download_wiki.py

Add description to dataset wiki

Bankso committed:
Create create_entity_links.py

Repurpose create_folder_ids to create link entities instead

Bankso committed:
Create create_entity_links.py

Repurpose create_folder_ids to create link entities instead

Bankso committed:
Update create_entity_links.py

Add link columns for tools and ed resources

Bankso committed:
Update create_entity_links.py

Add link columns for tools and ed resources

Bankso committed:
Update create_entity_links.py

Bankso committed:
Update create_entity_links.py

Bankso committed:
Update union_qc.py

Add debug option, which labels rows in "updated" output CSV with "Database" or "Updated", depending on the source of the row. This can help the user assess the difference between the current and updated entries, to see if there was an update made or if a typo was introduced during curation

Bankso committed:
Update union_qc.py

Add debug option, which labels rows in "updated" output CSV with "Database" or "Updated", depending on the source of the row. This can help the user assess the difference between the current and updated entries, to see if there was an update made or if a typo was introduced during curation

Bankso committed:
Update prep_backpopulation.py

- Added code to generate empty columns in manifests for new attributes
- Added logic to handle full and fragmented DOIs
- Added function to extract lists from columns (if needed)
- Added function to map old column names to new column names

Bankso committed:
Update prep_backpopulation.py

- Added code to generate empty columns in manifests for new attributes
- Added logic to handle full and fragmented DOIs
- Added function to extract lists from columns (if needed)
- Added function to map old column names to new column names

Bankso committed:
Update gen-mp-csv.py

Add logic to handle GrantView manifests

Bankso committed:
Update gen-mp-csv.py

Add logic to handle GrantView manifests

Bankso committed:
Update split_manifest_grants.py

Add logic to handle GrantView manifests

Bankso committed:
Update split_manifest_grants.py

Add logic to handle GrantView manifests

Bankso committed:
Update upload-manifests.py

Add option GrantView to data types

Bankso committed:
Update upload-manifests.py

Add option GrantView to data types

Bankso committed:
Update upload-manifests.py

Add option to skip validation and accompanying logic

Bankso committed:
Update upload-manifests.py

Add option to skip validation and accompanying logic

Bankso committed:
Update upload_validation.py

Add option to validate grant uploads

Bankso committed:
Update upload_validation.py

Add option to validate grant uploads

Bankso committed:
Update merge_tables.py

Add option to generate GrantView union tables

Bankso committed:
Update merge_tables.py

Add option to generate GrantView union tables

Bankso committed:
Update schema_update.py

Add column name patterns for Grant View schema

Bankso committed:
Update schema_update.py

Add column name patterns for Grant View schema

Bankso committed:
Create count_table_rows.py

Adds a simple utility script to count the number of rows present in a list of tables, based on their SynIds.

Bankso committed:
Create count_table_rows.py

Adds a simple utility script to count the number of rows present in a list of tables, based on their SynIds.

Bankso committed:
Create replace_table_schema.py

Add initial script to update table schemas for distributed grant-specific metadata stored in Synapse, based on schema_update.py.
The script removes old columns, adds new columns, and ensure columns are order properly. Old/new column lists are hardcoded and there is logic to select the correct column sets, depending on data type

Bankso committed:
Create replace_table_schema.py

Add initial script to update table schemas for distributed grant-specific metadata stored in Synapse, based on schema_update.py.
The script removes old columns, adds new columns, and ensure columns are order properly. Old/new column lists are hardcoded and there is logic to select the correct column sets, depending on data type

Bankso committed:
Update prep_backpopulation.py

Fix column order for publications

Bankso committed:
Update prep_backpopulation.py

Fix column order for publications

Bankso committed:
Update split_manifest_grants.py

Update grant number reference to GrantView Key, which will be used in manifests moving forward

Bankso committed:
Update split_manifest_grants.py

Update grant number reference to GrantView Key, which will be used in manifests moving forward

Bankso committed:
Update gen-mp-csv.py

Add information and logic to handle empty folder and project ids for Synapse projects that have more than one associated grant.

Bankso committed:
Update gen-mp-csv.py

Add information and logic to handle empty folder and project ids for Synapse projects that have more than one associated grant.

Bankso committed:
Update upload-manifests.py

Add logic to submit manifests one at a time. The option is coded under input arguments in main(), but could be moved to a command line argument

Bankso committed:
Update upload-manifests.py

Add logic to submit manifests one at a time. The option is coded under input arguments in main(), but could be moved to a command line argument

Bankso committed:
Update gen-mp-csv.py

Update logic to handle grant numbers that don't have correct synIds associated in gen-mp-csv outputs

Bankso committed:
Update gen-mp-csv.py

Update logic to handle grant numbers that don't have correct synIds associated in gen-mp-csv outputs

Bankso committed:
Create count_terms.py

Adapt count_table_rows.py to count term occurrences in CCKP database-style manifests, based on entries in all_valid_values.csv

Bankso committed:
Create count_terms.py

Adapt count_table_rows.py to count term occurrences in CCKP database-style manifests, based on entries in all_valid_values.csv

Bankso committed:
Update prep_backpopulation.py

Add some typing to help call proper functions

Bankso committed:
Update prep_backpopulation.py

Add some typing to help call proper functions

Bankso committed:
Update gen-mp-csv.py

Add Syn Id mappings for CA184898

Bankso committed:
Update gen-mp-csv.py

Add Syn Id mappings for CA184898

Bankso committed:
Update count_terms.py

Add logic to handle Tool-related valid value and count Input/Output columns for subset of Tool-related terms

Bankso committed:
Update count_terms.py

Add logic to handle Tool-related valid value and count Input/Output columns for subset of Tool-related terms

Bankso committed:
Update prep_backpopulation.py

Add logic and configurations to handle ToolView manifest processing

Bankso committed:
Update prep_backpopulation.py

Add logic and configurations to handle ToolView manifest processing

Bankso committed:
Update replace_table_schema.py

Add configuration to update Tool table schemas

Bankso committed:
Update replace_table_schema.py

Add configuration to update Tool table schemas

Bankso committed:
Update upload-workflow.sh

Use create_entity_links.py instead of create_id_folders.py

Bankso committed:
Update upload-workflow.sh

Use create_entity_links.py instead of create_id_folders.py

Bankso committed:
Update union_qc.py

Adjust column names and mappings to reflect updated models

Bankso committed:
Update union_qc.py

Adjust column names and mappings to reflect updated models

Bankso committed:
Update prep_backpopulation.py

Adjust column name sets to reflect updated models

Bankso committed:
Update prep_backpopulation.py

Adjust column name sets to reflect updated models

Bankso committed:
Update replace_table_schema.py

Adjust column names and mappings to reflect updated models

Bankso committed:
Update replace_table_schema.py

Adjust column names and mappings to reflect updated models

Bankso committed:
Update gen-mp-csv.py

Adjust logic to accept missing folder map

Bankso committed:
Update gen-mp-csv.py

Adjust logic to accept missing folder map

Bankso committed:
Update split_manifest_grants.py

Adjust logic to define primary key for ed resources at appropriate point in script

Bankso committed:
Update split_manifest_grants.py

Adjust logic to define primary key for ed resources at appropriate point in script

Bankso committed:
Update replace_table_schema.py

Remove entries from columns to add/drop, since resource table schemas have been updated to current model version

Bankso committed:
Update replace_table_schema.py

Remove entries from columns to add/drop, since resource table schemas have been updated to current model version

Bankso committed:
Update processing-splits.py

Add column ordering

Bankso committed:
Update processing-splits.py

Add column ordering

Bankso committed:
Update schema_update.py

Add additional columns to update

Bankso committed:
Update schema_update.py

Add additional columns to update

Bankso committed:
Update upload-manifests.py

Pass project Synapse Id to validation function

Bankso committed:
Update upload-manifests.py

Pass project Synapse Id to validation function

Bankso committed:
Update union_qc.py

Adjust for updated data models

Bankso committed:
Update union_qc.py

Adjust for updated data models

Bankso committed:
Update upload-manifests.py

Bankso committed:
Update upload-manifests.py

Bankso committed:
Update split_manifest_grants.py

Add logic to handle pubmed crawler manifests correctly

Bankso committed:
Update split_manifest_grants.py

Add logic to handle pubmed crawler manifests correctly

Bankso committed:
Update processing-splits.py

Add code to adjust Grant Number to GrantView Key and create a blank Study Key column

Bankso committed:
Update processing-splits.py

Add code to adjust Grant Number to GrantView Key and create a blank Study Key column

Bankso committed:
Update schema_update.py

Adjust attributes included in schema updates

Bankso committed:
Update schema_update.py

Adjust attributes included in schema updates

Bankso committed:
Update sync_publications.py

Bankso committed:
Update sync_datasets.py

Bankso committed:
Update sync_tools.py

Bankso committed:
Update sync_education.py

Bankso committed:
Update union_qc.py

Remove column filtering, since this column no longer exists in the database

Bankso committed:
Update union_qc.py

Remove column filtering, since this column no longer exists in the database

Bankso committed:
Update sync_publications.py

ignore errors when dropping columns

Bankso committed:
Update sync_datasets.py

Added logic to parse link and alias for Synapse-related patterns. Script will preferentially copy an existing Synapse link from the link column to the synapseLink column. If an appropriate link doesn't exist, the script will check if alias is a synId. If true, a new Synapse URL will be constructed and listed in synapseLink

Bankso committed:
Update sync_tools.py

Added logic to parse multiple tool links for Synapse-related patterns. Script will copy any existing Synapse links from the link columns to the synapseLink column. This can generate a list of links, in format [Link](Synapse URL)

Bankso committed:
Update sync_education.py

Added logic to parse link and alias for Synapse-related patterns. Script will preferentially check if alias is a synId. If true, a new Synapse URL will be constructed and listed in synapseLink. If the alias is not a synId, the script will check if the Resource Link is a Synapse URL and copy it to the synapseLink column, if true. Resource links will have the format [synId](URL) for alias match and [Link](URL) if a URL match

Bankso committed:
Update sync_publications.py

Drop unused columns and ignore errors when dropping columns

Bankso committed:
Update sync_datasets.py

Added new columns for CCKP table schema
Added code to drop unused columns
Added logic to parse link and alias for Synapse-related patterns. Script will preferentially copy an existing Synapse link from the link column to the synapseLink column. If an appropriate link doesn't exist, the script will check if alias is a synId. If true, a new Synapse URL will be constructed and listed in synapseLink

Bankso committed:
Update sync_tools.py

Updated columns to match UNION table content and CCKP table schema
Added code to drop unused columns
Added logic to parse multiple tool links for Synapse-related patterns. Script will copy any existing Synapse links from the link columns to the synapseLink column. This can generate a list of links, in format [Link](Synapse URL)

Bankso committed:
Update sync_education.py

Added columns for CCKP table schema
Added code to remove unused columns
Added logic to parse link and alias for Synapse-related patterns. Script will preferentially check if alias is a synId. If true, a new Synapse URL will be constructed and listed in synapseLink. If the alias is not a synId, the script will check if the Resource Link is a Synapse URL and copy it to the synapseLink column, if true. Resource links will have the format [synId](URL) for alias match and [Link](URL) if a URL match

Bankso committed:
Merge branch '159-doi-download-link-curation' into 2-2025-updates

Bankso committed:
Update synapse_json_schema_bind.py

Add code to bind AR schema with derived annotations enabled and adjust naming to ensure proper URI is used if schema has been previously registered

Bankso committed:
Create build_tag_lists.py

Bankso committed:
Squashed commit of the following:

commit 7691b4279f612c9e5b7b90343ecaef900de4aede
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Fri Feb 21 16:37:17 2025 -0800

    Update synapse_json_schema_bind.py

    Add code to bind AR schema with derived annotations enabled and adjust naming to ensure proper URI is used if schema has been previously registered

Bankso committed:
Squashed commit of the following:

commit 411c94a6c03deeb3d2b4839f2278974a9c7fdcc7
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 15:55:57 2025 -0800

    Create build_tag_lists.py

commit 7691b4279f612c9e5b7b90343ecaef900de4aede
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Fri Feb 21 16:37:17 2025 -0800

    Update synapse_json_schema_bind.py

    Add code to bind AR schema with derived annotations enabled and adjust naming to ensure proper URI is used if schema has been previously registered

Bankso committed:
Update build_tag_lists.py

Restructure loops to collect all annotations, then compare to list, to ensure tags are not overwritten. Script tested with two input columns and produces expected result.

Bankso committed:
Squashed commit of the following:

commit a0e98886325b5c1a203467527743b2f8a89bc259
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Wed Mar 5 12:31:40 2025 -0800

    Update sync_tools.py

    adjust df.rename input

commit b5fc6573bdf04ba18c9016c42eb3ac0fb00861b0
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Wed Mar 5 12:30:16 2025 -0800

    Update sync_publications.py

    adjust df.rename input

commit 1ed981ce8e3cdb16716e00663d49bfd23f2d88cf
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Wed Mar 5 12:16:40 2025 -0800

    Update sync_education.py

    Cast entries to pandas series to enable conversion to lists. addresses error `AttributeError: 'DataFrame' object has no attribute 'str'.` when trying to run function utils.convert_to_stringlist(). Functions expects series but other changes were resulting in a dataframe object being passed instead

commit 93304978ca203e87b49b5de85d0225d095a29ad6
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:51:19 2025 -0800

    Update sync_education.py

    adjust order to prefer building url from alias and reformat list comprehension for url capture

commit 603c63d15cf45fae793e65051b07a8c20b8555e0
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:42:24 2025 -0800

    Update sync_education.py

    Condense alias and url extraction

commit dcc267c488e47c8aa6109e8a24925c808a325a32
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:03:27 2025 -0800

    Update sync_tools.py

    rename columns instead of adding and dropping

commit f22ea9ec5fe1b93f9cedfa1a27e54e9b9a2c166d
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:03:22 2025 -0800

    Update sync_publications.py

    rename columns instead of adding and dropping

commit c4005f4d6908122fab546fe6ca55386d484461b3
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:03:08 2025 -0800

    Update sync_education.py

    rename columns instead of adding and dropping

commit 5b0a08b8fcb38e1d8c622f0affb34e9712170565
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 17:02:49 2025 -0800

    Update sync_datasets.py

    rename columns instead of adding and dropping

commit 7e9218aaecdc1a78dfa1557dea5d775515b8ca50
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 16:45:02 2025 -0800

    Update portal_tables/sync_tools.py

    Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

commit 6fe64eb44625c8e28c970b140a3c5d9d27a6e7de
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 16:44:02 2025 -0800

    Update portal_tables/sync_datasets.py

    Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

commit 6c8cb018fad5e31821fa3c1841427df8327ee94f
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 16:43:19 2025 -0800

    Update portal_tables/sync_datasets.py

    Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

commit 063f338e29dd87fd19ad20d948dcd3d027959f85
Author: Orion Banks <49208907+Bankso@users.noreply.github.com>
Date:   Tue Mar 4 16:42:12 2025 -0800

    Update portal_tables/sync_datasets.py

    Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

Bankso committed:
Create build_tag_lists.py

Script takes inputs:
- metadata CSV with relevant annotations (intended to be used as part of syncing scripts)
- term CSV with columns "term" and "label". Term contains annotations found in the metadata CSV, label contains the categorical terms that will be used for icons on the CCKP. This list is maintained in the data-models repo as modules/shared/mc2_iconTag_map_3-4-25.csv
- annotation types analyzed for tags are defined by attribute_list in main on row 78

Bankso committed:
Add "iconTags" to sync scripts

"iconTags" added for conversion to string list and inclusion in final database CSVs

Bankso committed:
Merge branch '159-part2-iconTags' into portal-icon-tags

Bankso committed:
Merge branch 'portal-icon-tags' into 2-2025-updates

Bankso committed:
Update processing-splits.py

Modify input to read the "File Paths" column from gen-mp-csv.py output CSV

Bankso committed:
Delete synapse_json_schema_bind.py

Bankso committed:
Revert "Delete synapse_json_schema_bind.py"

This reverts commit 359585807b8c51f994c2f53734af9c11c8202c04.

Bankso committed:
Update create_grant_projects.py

Rename attribute references to fit current GrantView table and add code to print new team and project information to terminal

Bankso committed:
Merge branch 'main' into 2-2025-updates

Bankso committed:
Update sync_datasets.py

Remove 'synapseLink' extraction from sync_datasets.py

Bankso committed:
Update sync_datasets.py

Adjust column ordering to fit portal table schema

Bankso committed:
Merge branch 'update-portal-sync' into 2-2025-updates

Bankso committed:
Update build_tag_lists.py

Remove logic to select sep - since this script will be run before the portal sync scripts, all column names will contain a space

Bankso committed:
Update schema_update.py

Add additional Tool View columns to be updated

Bankso committed:
Update schema_update.py

Update column names

Bankso committed:
Update build_datasets.py

Add functionality to select files of given format(s) when creating datasets

aditya-nath-sage committed:
Boolean column script prototype

aditya-nath-sage commented:
Added the boolean column script work to this PR: https://github.com/mc2-center/mc2-center-dcc/pull/82/commits/bb961d8d468b752d2b4df8458cfde8a1e11ab026

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Create table_to_annotations.py

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Update table_to_annotations.py

Bankso committed:
Merge branch '153-apply-record-annotations' into 2-2025-updates

Bankso committed:
Update synapse_json_schema_bind.py

Bankso committed:
Update synapse_json_schema_bind.py

Bankso committed:
Update build_datasets.py

Bankso committed:
Update build_datasets.py

Bankso committed:
Merge branch 'main' into 2-2025-updates

Bankso committed:
Update build_datasets.py

Bankso committed:
Create build_datasets.py

Bankso committed:
Merge branch 'add-build-datasets' into 2-2025-updates

Bankso committed:
Update table_to_annotations.py

Implement inputs, logic, and additional function to apply DatasetView annotations to Dataset entities

Bankso committed:
Merge branch 'annotate-datasets' into 2-2025-updates

Bankso committed:
Update table_to_annotations.py

Add argument for Dataset Synapse Id to collect_dataset_annotations, which is used to select table rows and apply annotations

Bankso committed:
Merge branch 'annotate-datasets' into 2-2025-updates

aditya-nath-sage committed:
Added CODEOWNERS file to make sure I am tagged for all changes to this repo, along with @Bankso for now
